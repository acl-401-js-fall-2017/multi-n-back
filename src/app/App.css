.App {
  text-align: center;
  min-height: 100%;
  min-width: 100%;
  position: relative;
}


.home > a > button {
  width: 70%;
  margin-top: 0.8em;
  border-radius: 0.4em;
  border: none;
  padding: 0.5em 2em 0.5em 2em;
  background: #568289;
  color: #d8eaec;
  font-family: 'Gruppo', cursive;
  font-size: 100%;
  font-weight: bold;
}

.home > a > button:hover {
  background-image: -webkit-gradient(radial, center, from(#d8eaec), to(end-color));
  background-image: -webkit-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
  background-image: -moz-radial-gradient(center, #d8eaec, #ffffff00 25%, #ffffff00 75%, #d8eaec);
  background-image: radial-gradient(at center, #ffffff00 60%, #d8eaec);
} 

footer {
  width: 100%;
  height: 10em;
  margin-top: 4em;
  /* background-color: #254044; */
  position: relative;
}

footer::after {
  content: '';
  position: absolute;
  z-index: -1;
  height: 720px;
  top: 0; right: 0; bottom: 0; left: 0;
  background-image: -webkit-gradient(linear, left bottom, left top, from(#d8eaec), to(white));
  background-image: -webkit-linear-gradient(bottom, #d8eaec, white);
  background-image: -moz-linear-gradient(bottom, #d8eaec, white);
  background-image: linear-gradient(to top, #d8eaec00 80%, white);
}

footer::before {
  content: '';
  position: absolute;
  z-index: -100;
  height: 720px;
  background-size: cover;
  background-attachment: fixed;
  background-color: #254044;
  top: 0; right: 0; bottom: 0; left: 0;
}
