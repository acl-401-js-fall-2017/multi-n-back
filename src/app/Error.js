import React from 'react';
import { ErrorDiv } from '../styles/style';

export default ({ error }) => {
  if(!error) return null;

  return (
    <ErrorDiv>
      {Array.isArray(error) 
        ? <ul>error.map(err => <li>err</li>)</ul>
        : error.error ? error.error : error
      }
    </ErrorDiv>
  );
};