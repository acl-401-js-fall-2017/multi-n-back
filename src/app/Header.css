/*
japanese indigo: #254044
barn red: #770d00
wintergreen dream: #568289
azureish white: #d8eaec
white: #fff
*/


.hero-header {
    height: 720px;
    position: relative;
    z-index: 0;
}

.hero-header::after {
    content: '';
    position: absolute;
    z-index: -1;
    height: 720px;
    top: 0; right: 0; bottom: 0; left: 0;
    background-image: -webkit-gradient(linear, left top, left bottom, from(#d8eaec), to(white));
    background-image: -webkit-linear-gradient(top, #d8eaec, white);
    background-image: -moz-linear-gradient(top, #d8eaec, white);
    background-image: linear-gradient(to bottom, #d8eaecbe 60%, white);
}

.hero-header::before {
    content: '';
    position: absolute;
    z-index: -100;
    height: 720px;
    background-size: cover;
    background-attachment: fixed;
    background-image: url(https://d2v9y0dukr6mq2.cloudfront.net/video/thumbnail/Vio80dDve/neuron-network-activity-background-animation-for-a-biological-structure-and-its-network-communication-activity_rpuopx28__F0000.png);
    top: 0; right: 0; bottom: 0; left: 0;
}

.hero-header h1 {
    margin: 0;
}

.hero-header .auth {

}

.hero-header .nav {

}
















.collapsed-header {
    height: 80px;
    overflow: scroll;
}

.collapsed-header h1 {

}

.collapsed-header .auth {

}

.collapsed-header .nav {

}
