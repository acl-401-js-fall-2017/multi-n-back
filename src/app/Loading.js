import React from 'react';
import { LoadingStyled } from '../styles/style';

export default ({ loading }) => {
  if(!loading) return null;

  return (
    <LoadingStyled>
      Loading...
    </LoadingStyled>
  );
};