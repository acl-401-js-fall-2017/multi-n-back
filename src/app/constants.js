// ###  Auth Constants ### //
export const SIGN_IN = 'SIGN_IN';
export const LOG_OUT = 'LOG_OUT';
export const CHECKED_TOKEN = 'CHECKED_TOKEN';
export const GOT_TOKEN = 'GOT_TOKEN';
export const AUTH_ERROR = 'AUTH_ERROR';

// ###  Game Constants ### //
export const INIT_GAME = 'INIT_GAME';
export const SET_SETTINGS = 'SET_SETTINGS';
export const INIT_SEQUENCE = 'INIT_SEQUENCE';
export const NEXT_COMBO = 'NEXT_COMBO';
export const SEQUENCE_OVER = 'SEQUENCE_OVER';
export const GAME_OVER = 'GAME_OVER';
export const WRAP_UP = 'WRAP_UP';
export const LEAVE_GAME = 'LEAVE_GAME';

// ### Stats ### //
export const GET_COMMUNITY_STATS = 'GET_COMMUNITY_STATS';
export const GET_USER_STATS = 'GET_USER_STATS';

// ### Loading / Error Constants ### //
export const LOADING = 'LOADING';
export const LOADED = 'LOADED';
export const ERROR = 'ERROR';