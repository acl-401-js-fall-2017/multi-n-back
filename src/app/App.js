import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Home from '../home/Home';
import Header from './Header';
import About from '../about/About';
import Game from '../game/Game';
// import Error from './Error';
// import Loading from './Loading';
import PrivateRoute from '../auth/PrivateRoute';

import { connect } from 'react-redux';

import './App.css'


class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Header/>
          <Switch>
            <Route exact path="/" component={Home}/>
            <PrivateRoute path="/about" component={About}/>
            {/* <Route path="/game" component={Game}/> */}
            <PrivateRoute path="/game" component={Game}/>
          </Switch>
          <footer>
            <h1>FOOOOOOOOOOOOOT</h1>
          </footer>
        </div>
      </Router>
    );
  }
}

export default connect(
  state => ({ 
    loading: state.loading,
    error: state.error
  }),
  null
)(App);
