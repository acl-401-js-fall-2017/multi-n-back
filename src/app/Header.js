import React, { Component } from 'react';
import { connect } from 'react-redux';

import Auth from '../auth/Auth';
import Nav from '../nav/Nav';

import './Header.css';

class Header extends Component {
  render() {
    const { user } = this.props;

    return (
      <header className={user ? 'collapsed-header' : 'hero-header'}>
        <h1>multi N back</h1>
        <Auth/>
        <Nav/>
      </header>  
    );
  }
}
  
export default connect(
  state => ({
    user: state.user
  }),
  null
)(Header);
  