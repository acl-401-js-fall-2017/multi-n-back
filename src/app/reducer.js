import * as actions from './constants';

export function loading(state = false, { type }) {

  switch(type) {
    case actions.LOADING:
      return true;
    case actions.LOADED:
    case actions.ERROR:
      return false;
    default:
      return state;
  }
}

export function error(state = null, { type, payload }) {

  switch(type) {
    case actions.ERROR:
      return payload;
    case actions.LOADING:
      return null;
    default:
      return state;
  }
}