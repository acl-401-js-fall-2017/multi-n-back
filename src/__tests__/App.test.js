import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../app/App';

describe('App', () => {
  it('shows App component', () => {
    const checkForToken = () => {};
    const wrapper = shallow(
      <App checkForToken={checkForToken}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});