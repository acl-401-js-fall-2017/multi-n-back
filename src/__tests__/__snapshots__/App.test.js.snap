// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shows App component 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Connect(Header) />
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        path="/game"
      />
    </Switch>
    <footer />
  </div>
</BrowserRouter>
`;
